### PR TITLE
Project structure: added lazy loading modules

### PR DESCRIPTION
**Pull request by trotsm**
Created project structure: added lazy loading modules (login, shell, student-book, progress, journal, admin-panel) with components. Added HashLocationStrategy.

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  {
+    path: 'login',
+    loadChildren: './login/login.module#LoginModule'
+  },
+  {
+    path: 'shell',
+    loadChildren: './shell/shell.module#ShellModule'
+  },
+  {
+    path: 'admin-panel',
```
Comment (dmitryy90) [inline src/app/app-routing.module.ts]: app router should have only login and shell router

other routers need to be moved to shell router

Comment (trotsm) [reply src/app/app-routing.module.ts]: Done!

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  {
+    path: 'login',
+    loadChildren: './login/login.module#LoginModule'
+  },
+  {
+    path: 'shell',
+    loadChildren: './shell/shell.module#ShellModule'
+  },
+  {
+    path: '',
```
Comment (dmitryy90) [inline src/app/app-routing.module.ts]: try to make shell component as root route

Comment (trotsm) [reply src/app/app-routing.module.ts]: Done!

**Pull request merged**